lib/routes/veeam/get.js

```diff
@@ -25,21 +26,44 @@ function getVeeamFile(request, response, bucketMd, log) {
         if (err) {
             return responseXMLBody(errors.InternalError, null, response, log);
         }
-        const fileToBuild = getFileToBuild(request, data._capabilities?.VeeamSOSApi);
-        if (fileToBuild.error) {
-            return responseXMLBody(fileToBuild.error, null, response, log);
-        }
-        let modified = new Date().toISOString();
-        // Extract the last modified date, but do not include it when computing
-        // the file's ETag (md5)
-        modified = fileToBuild.value.LastModified;
-        delete fileToBuild.value.LastModified;
+        const finalizeRequest = bucketMetrics => {
+            const fileToBuild = getFileToBuild(request, data._capabilities?.VeeamSOSApi);
+            if (fileToBuild.error) {
+                return responseXMLBody(fileToBuild.error, null, response, log);
+            }
+            let modified = new Date().toISOString();
+            // Extract the last modified date, but do not include it when computing
+            // the file's ETag (md5)
+            modified = fileToBuild.value.LastModified;
+            delete fileToBuild.value.LastModified;
+
+            // The SOSAPI metrics are dynamically computed using the SUR backend.
+            if (bucketMetrics && !fileToBuild.value.CapacityInfo?.Used) {
+                fileToBuild.value.CapacityInfo.Used = Number(bucketMetrics.bytesTotal);
+                fileToBuild.value.CapacityInfo.Available =
+                    Number(fileToBuild.value.CapacityInfo.Capacity) - Number(bucketMetrics.bytesTotal);
+                // TODO when SUR backend supports realtime metrics: enable this, it will report the
+                // real last modified date of the bucket, instead of the current date, ensuring
+                // no issue in a SOSAPI context.
```

Comment:
should have a ticket linked to this ?

Reply:
I added one, but it might change or we might forget it for some time, but at least we have a ticket number with context...